File: python/mpi4py/hello.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

# Copyright (c) 2013 Jérémie DECOCK (http://www.jdhp.org)

# run:
#   mpirun -np 4 python3 hello.py
#     or
#   mpiexec -n 4 python3 hello.py

from mpi4py import MPI

rank = MPI.COMM_WORLD.rank
size = MPI.COMM_WORLD.size
name = MPI.Get_processor_name()

print("process %d of %d on %s" % (rank, size, name))

File: cpp/boost_mpi/hello.cc
#include <boost/mpi/environment.hpp>
#include <boost/mpi/communicator.hpp>

namespace mpi = boost::mpi;

/*
 * Required packages (Debian): mpi-default-bin, mpi-default-dev, libboost-mpi-dev
 *
 * Compile:
 *   mpic++ -o hello hello.cc -lboost_mpi
 *
 * Run:
 *   mpirun -np 4 hello
 * 
 * See:
 * - https://www.boost.org/doc/libs/1_68_0/doc/html/mpi/tutorial.html
 * - https://www.boost.org/doc/libs/1_68_0/doc/html/mpi/getting_started.html
 * - https://stackoverflow.com/questions/40568713/unable-to-compile-simple-boost-mpi-example
 */

int main()
{
    mpi::environment env;
    mpi::communicator world;

    std::cout << "I am process " << world.rank() << " of " << world.size() << "." << std::endl;
    
    return 0;
}